enum.rb

```diff
@@ -0,0 +1,134 @@
+# frozen_string_literal: true
+
+module Enumerable
+  def my_each
+    return to_enum unless block_given?
+
+    i = 0
+    if block_given?
+      while i < size
+        yield(self[i])
+        i += 1
+      end
+    end
+  end
+
+  def my_each_with_index
+    return to_enum unless block_given?
+
+    i = 0
+    if block_given?
+      while i < size
+        yield(self[i], i)
+        i += 1
+      end
+    end
+  end
+
+  def my_select
+    return to_enum unless block_given?
+
+    i = 0
+    new_array = []
+    while i < size
+      new_array << self[i] if yield(self[i]) == true
+      i += 1
+    end
+    new_array
+  end
+
+  def my_all?(arg = nil)
+    if block_given?
+      my_each { |x| return false unless yield(x) }
+    elsif arg.class == Class
+      my_each { |x| return false unless x.class == arg }
+    elsif arg.class == Regexp
+      my_each { |x| return false if (x =~ arg).nil? }
+    elsif !arg.nil?
+      my_each { |x| return false unless x == arg }
+    elsif arg.nil?
+      my_each { |x| return false unless x }
+    end
+    true
+  end
+
+  def my_any?(arg = nil)
+    if block_given?
+      my_each { |x| return false unless yield(x) }
+    elsif arg.class == Class
```

Comment:
[OPTIONAL]
a simple `===` would cover **Class**, **RegExp** and **Pattern**
All of the extra ifs not necessary. Only need 3
For block, when arg not nil and when arg nil and block not given.